packages/experience-builder-sdk/src/blocks/editor/VisualEditorContext.tsx

```diff
@@ -113,40 +116,47 @@ export function VisualEditorContextProvider({
         locale: locale,
       })
     );
+    setEntitiesFetched(false);
   }, [entityStore, locale]);
 
   // Either gets called when dataSource changed or designComponetsRegistry changed (manually)
-  const fetchMissingEntities = useCallback(async () => {
-    const entityLinks = [...Object.values(dataSource), ...designComponentsRegistry.values()];
-    const { missingAssetIds, missingEntryIds } = entityStore.getMissingEntityIds(entityLinks);
-    // Only continue and trigger rerendering when we need to fetch something and we're not fetching yet
-    if (!missingAssetIds.length && !missingEntryIds.length) {
-      setInitialEntitiesFetched(true);
-      return;
-    }
-    setInitialEntitiesFetched(false);
-    setFetchingEntities(true);
-    try {
-      // Await until the fetching is done to update the state variable at the right moment
-      await entityStore.fetchEntities({ missingAssetIds, missingEntryIds });
-      console.debug('[exp-builder.sdk] Finished fetching entities', { entityStore, entityLinks });
-    } catch (error) {
-      console.error('[exp-builder.sdk] Failed fetching entities');
-      console.error(error);
-    } finally {
-      // Important to set this as it is the only state variable that triggers a rerendering
-      // of the components (changes inside the entityStore are not part of the state)
-      setInitialEntitiesFetched(true);
-      setFetchingEntities(false);
-    }
-  }, [dataSource, entityStore]);
+  const fetchMissingEntities = useCallback(
+    async (newDataSource?: typeof dataSource) => {
```

Comment:
why not directly `CompositionDataSource` type?

Reply:
Huh, dunno why I did this as the type is already imported and quite simple to write out. Will replace it 👍